docs/turn-howto.md

```diff
@@ -96,10 +109,28 @@ This will install and start a systemd service called `coturn`.
         # TLS private key file
         pkey=/path/to/privkey.pem
 
+    In this case, replace the `turn:` schemes in the `turn_uri` settings below
+    with `turns:`.
+
+    We recommend that you only try to set up TLS/DTLS once you have set up a
+    basic installation and got it working.
+
 1.  Ensure your firewall allows traffic into the TURN server on the ports
-    you've configured it to listen on (By default: 3478 and 5349 for the TURN(s)
-    traffic (remember to allow both TCP and UDP traffic), and ports 49152-65535
-    for the UDP relay.)
+    you've configured it to listen on (By default: 3478 for TURN and 5349 for
+    TURNs traffic (remember to allow both TCP and UDP traffic), and ports
+    49152-65535 for the UDP relay.)
+
+1.  If your TURN server is behind NAT, you will need to tell it its external IP
+    address:
+
+        external-ip=192.88.99.1
+
+    For this to work, your NAT gateway must forward all of the relayed ports
+    directly (relayed port 12345 must be always mapped to the same 'external'
+    port 12345).
+
+    We are not aware of anyone who has successfully configured a TURN server
+    behind NAT. If you get it working, let us know!
```

Comment:
It fees like we should be more actively discouraging running turn servers behind NATs? I'm not sure if it'll work if the client is behind a particularly aggressive NAT too?

Reply:
welll... I think it *should* work, if the NAT server is correctly configured, so I'm reluctant to say "thou shalt absolutely not do this". I was hoping for a compromise which basically amounts to: if you want to try this, you're on your own and you better be a networking guru.

Reply:
Running a service to punch holes through NATs behind a NAT feels like it might turn out to be painful TBH. Can we maybe start the section with "We are not aware of anyone successfully running a TURN server behind a NAT. However, if you're attempting to then you'll need to at least do the following...", or something. Perhaps also pointing at the coturn docs?

Reply:
> Running a service to punch holes through NATs behind a NAT feels like it might turn out to be painful TBH.

well yes, but plenty of people try it.

coturn's docs are... not terribly accessible here.

I'll reshuffle the section as you suggest.

Reply:
hopefully better now?

Reply:
Just wanted to let you know I followed the instructions and got it working successfully behind a NAT on my first try, no issue whatsoever. I just forwarded the mentioned ports to the local IP running the turn server, opened the same ports to incoming traffic on that machine, and it just worked...

Maybe it's just me, but this warning seems to scare people off unnecessarily for no good reason.

Reply:
yes, I'd be happy for us to downgrade this warning to "it's an advanced topic". I'm glad to hear that you got it working, but the fact you did suggests that you have slightly more networking experience than the average person that tries to set up synapse :)

anyway if you'd like to send a PR to update the doc, I'd be glad to look at it :)

Reply:
>but the fact you did suggests that you have slightly more networking experience than the average person that tries to set up synapse :)

I would argue that anyone that tries to setup anything behind a NAT needs to have an above average network experience, it's nothing specific to a turn server :wink:

Reply:
> I would argue that anyone that tries to setup anything behind a NAT needs to have an above average network experience

well this is rather the point. If you *don't* have above average network experience, please don't start by complaining to us that your TURN doesn't work ;)

In any case: TURN is definitely harder than averabe, on account of needing to map *all* of the external ports to the *same* internal ports, and then getting the "public IP address" right. My experience is that getting TURN working reliably is a black art at the best of times. Throwing NAT into the mix adds a whole lot more complication.